From patchwork Wed Apr 10 09:55:21 2019
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Paolo Bonzini <pbonzini@redhat.com>
X-Patchwork-Id: 10893557
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id D48E4139A
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 10 Apr 2019 09:55:43 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id BF277266F3
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 10 Apr 2019 09:55:43 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id B38002880C; Wed, 10 Apr 2019 09:55:43 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.7 required=2.0 tests=BAYES_00,DKIM_INVALID,
	DKIM_SIGNED,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 61B60266F3
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 10 Apr 2019 09:55:43 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1730101AbfDJJzZ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 10 Apr 2019 05:55:25 -0400
Received: from mail-wm1-f67.google.com ([209.85.128.67]:51766 "EHLO
        mail-wm1-f67.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726630AbfDJJzZ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 10 Apr 2019 05:55:25 -0400
Received: by mail-wm1-f67.google.com with SMTP id 4so1916180wmf.1;
        Wed, 10 Apr 2019 02:55:23 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=sender:from:to:subject:date:message-id;
        bh=aKeSTMQjglWGWQ91852+W2B2sqS7Gn4OwXBiExaxOTE=;
        b=SBIfW8BjpdnKEttHAx7OWihVAqLr6TIJYpgGWBzR/amL0NZaJa681VixCO8pc2wWiG
         wfRdQjhFImTNdLzhyjuOfcQFTR7Ksvf14XuSEoWywtIgWa7daVhZoGkUXrABusvPFhfg
         QwYlFFMc2cxX6JEd1muRCzJILSO6rUb3aHw46rG2w7ut2AN+Qc5pfWoeQi7NmqMKTEjX
         ozTKa12MU/nxXsgciEngqJG2Zufz6H22+8on1YamlAq+Vvnu2vccuyN0h61zVT1CdWVx
         SXn4wO/sZc17NcoUiwjfWG4Ia0P6rhsbTqFw0y9yUYjoJ9/TcA7CNMJ/UpK66W2lKa3M
         5kDA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:sender:from:to:subject:date:message-id;
        bh=aKeSTMQjglWGWQ91852+W2B2sqS7Gn4OwXBiExaxOTE=;
        b=bDz/TVFDlBGhvnDg4HrjFnbisZuo60VenEVXy4dN5tc1a1EEaIIsoJvwBm9JUlfRN2
         9sP0up9I4SYf2unV4JgHobuif3osNBLnYqDwcIEqGcp6XCky27gvw3WW0JXkynIkc+Pg
         tGPC1jAZhBtKligFtfuM9xjM8vsU4we68XcB0uZB4sgIILstmKBxbzy2XyTZaTM/XFOP
         MS3bcv1exd5z6YJGPjf7fGOIaTY9cQUvmuGTytiyXVBqXnXrs9ALNpD/bRQb6dngdNvU
         78u5bjY0Vym9yiLj7czxDBSJeq0jgaOnvLj+cf21an+tiqDaqXyUHIhyT5KVYQ9ftvBb
         RW2Q==
X-Gm-Message-State: APjAAAVxdmTOJpKYscCYFMt7jgYw8O75S6y+wH/bNdDKJPsoHEnXhUnd
        HXZSd/ujLX/S84Yoexo9aCF2XVhT
X-Google-Smtp-Source: 
 APXvYqwxdyleq6ClHxOFvX4yWIOutyO+WKk9c9LyezSXA6qK6+Qii5rZ/ejntCbHknvJfmgbtL8adA==
X-Received: by 2002:a05:600c:211a:: with SMTP id
 u26mr2259684wml.74.1554890122791;
        Wed, 10 Apr 2019 02:55:22 -0700 (PDT)
Received: from 640k.lan ([93.56.166.5])
        by smtp.gmail.com with ESMTPSA id
 z18sm54965915wrr.90.2019.04.10.02.55.21
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Wed, 10 Apr 2019 02:55:22 -0700 (PDT)
From: Paolo Bonzini <pbonzini@redhat.com>
To: linux-kernel@vger.kernel.org, kvm@vger.kernel.org
Subject: [PATCH] KVM: x86: clear VM_EXIT_SAVE_IA32_PAT
Date: Wed, 10 Apr 2019 11:55:21 +0200
Message-Id: <1554890121-57272-1-git-send-email-pbonzini@redhat.com>
X-Mailer: git-send-email 1.8.3.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

This is not needed, PAT writes always take an MSR vmexit.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Reviewed-by: Sean Christopherson <sean.j.christopherson@intel.com>
---
 arch/x86/kvm/vmx/vmx.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/arch/x86/kvm/vmx/vmx.c b/arch/x86/kvm/vmx/vmx.c
index 687d80cca09f..b6c533afbf27 100644
--- a/arch/x86/kvm/vmx/vmx.c
+++ b/arch/x86/kvm/vmx/vmx.c
@@ -2288,7 +2288,6 @@ static __init int setup_vmcs_config(struct vmcs_config *vmcs_conf,
 	min |= VM_EXIT_HOST_ADDR_SPACE_SIZE;
 #endif
 	opt = VM_EXIT_LOAD_IA32_PERF_GLOBAL_CTRL |
-	      VM_EXIT_SAVE_IA32_PAT |
 	      VM_EXIT_LOAD_IA32_PAT |
 	      VM_EXIT_LOAD_IA32_EFER |
 	      VM_EXIT_CLEAR_BNDCFGS |
